### DISCUSSION TITLE
Firmware readout and patching utility?

### DISCUSSION THREAD

**Discussion by Sabrehawk** [liquidctl/liquidctl#352] [Ideas]
Under Windows is there a way to extract/backup the firmware and update it manually without the dogpoop that is ICUE ? (Which is basically malware in my opinion...)

I guess there is not....i couldnt find anything.

But ..would it be very complicated to make a simple command line patcher for the HYDRO 115 i  ? 

its this particular (first?) version of the h115 (no pros, no xts, no plats)
├── Vendor ID: 0x1b1c
├── Product ID: 0x0c0a
├── Release number: 0x0100
├── Serial number: xxxxxxxxx (redacted)
├── Bus: usb2
├── Address: 3
├── Port: 11
└── Driver: Hydro690Lc using module usb

**Comment by jonasmalacofilho** [comment]
Based on what I've seen with other devices, it probably isn't very complicated.  But it's possible that any bug in the process bricks the device, making it something painful to get right.

And the biggest issue I see is regarding what firmware you would install with this tool.  We can't distribute the firmware files ourselves due to copyright (so the end user would need iCue anyway to install and official firmware).  I also haven't seen patched or open firmware blobs floating on the internet.  Are you planning to patch the firmware yourself?

**Comment by Sabrehawk** [comment]
I dont even know if theres a more current  one than i have (i doubt it) but i simply refuse to install ICUE ever again because its HORRIBLE, it tanks cpu cycles like crazy even when idle, its almost like im running some low level crypto miner on my system, it also constantly breaks on windows dev updates. ..and its "uninstall" is more than dirty.
I was basically just hoping for a clean, minimal solution..guess im out of luck...wonder how Linux users do it.

**Reply by jonasmalacofilho** [reply]
> wonder how Linux users do it.

Personally, using the manufacturer provided tool, either through a VM or by dual booting into Windows.  Not ideal, but so far I haven't been able to justify to myself spending a lot of time for something I will probably only use a couple of times throughout the life of the cooler.  And, like I mentioned before, I would still need the official tool anyway, to extract the firmware blob from it...

I think working on this feature makes a lot more sense when the goal is running partially or entirely custom firmware.  I have some interest in working on that latter part (and, consequently, on making the firmware update tool too), but it's not that high on my "cool ideas" list: building my own controller from scratch would probably be easier and more interesting.
